### PR TITLE
Distinct and Aggregate support

### PR DESCRIPTION
**Pull request by SaikiranDaripelli**
I am addind support for distinct(http://mongodb.github.io/node-mongodb-native/2.0/api/Collection.html#distinct) and aggregate(http://mongodb.github.io/node-mongodb-native/2.0/api/Collection.html#aggregate)

### CODE REVIEW

**Comment by SaikiranDaripelli** [comment]
This pull request should address https://github.com/vdemedes/mongorito/issues/107 , Please merge .

File: lib/mongorito.js [hunk 1]
```diff
@@ -753,6 +753,36 @@ Model.count = function (query) {
...
+ *
+ * @param {String} field for distinct
+ * @param {Object} query - query to filter the results
+ * @see http://mongodb.github.io/node-mongodb-native/2.0/api/Collection.html#distinct
+ * @api public
+ */
+
+Model.distinct = function (field, query) {
+	// collection, model
+	return new Query(this._collection(), this).distinct(field, query);
```
Comment (vdemedes) [inline lib/mongorito.js]: Missing indentation.

File: lib/mongorito.js [hunk 2]
```diff
@@ -753,6 +753,36 @@ Model.count = function (query) {
...
+ * Aggregation query
+ *
+ * @param {String} pipeline aggregation pipeline
+ * @param {Object} options - Options to be passed to aggregation pipeline
+ * @see http://mongodb.github.io/node-mongodb-native/2.0/api/Collection.html#distinct
+ * @api public
+ */
+
+Model.aggregate = function (pipeline) {
+	return new Query(this._collection(), this).aggregate(pipeline);
```
Comment (vdemedes) [inline lib/mongorito.js]: Missing indentation.

File: test/queries.js [hunk 3]
```diff
@@ -446,3 +446,42 @@ test('search documents using text index', async t => {
 	t.is(posts[0].get('title'), 'San Francisco');
 	t.is(posts[1].get('title'), 'San Fran');
 });
+
+test('get distinct', async t => {
+	try {
```
Comment (vdemedes) [inline test/queries.js]: Missing indentation.

File: lib/query.js [hunk 4]
```diff
@@ -184,6 +184,46 @@ Query.prototype.search = function (text) {
...
+ * Aggregation query
+ *
+ * @param {String} pipeline aggregation pipeline
+ * @param {Object} options - Options to be passed to aggregation pipeline
+ * @see http://mongodb.github.io/node-mongodb-native/2.0/api/Collection.html#aggregate
+ * @api public
+ */
+
+Query.prototype.aggregate = function (pipeline) {
+	return this.collection.then(function (collection) {
```
Comment (vdemedes) [inline lib/query.js]: Indentation again.

File: test/queries.js (hunk 3 above)
Comment (SaikiranDaripelli) [inline test/queries.js]: What indentation is missing? also npm test passed checking style guide defined, is there any thing extra i need to check?

File: lib/query.js (hunk 4 above)
Comment (SaikiranDaripelli) [inline lib/query.js]: What indentation is missing? also npm test passed checking style guide defined, is there any thing extra i need to check?

**Comment by vdemedes** [comment]
Oh, sorry, GitHub's diff ate the indentation for some reason.

**Comment by vdemedes** [comment]
Thank you, merged!

File: lib/mongorito.js (hunk 1 above)
Comment (SaikiranDaripelli) [inline lib/mongorito.js]: I am not sure why indentation is not showing up in diff, but if you view file, you can see that return in one tab indented same as other methods in the file.

File: lib/mongorito.js (hunk 2 above)
Comment (SaikiranDaripelli) [inline lib/mongorito.js]: Same as above

**Comment by SaikiranDaripelli** [comment]
Thanks

**Pull request merged**